File: rudyAir/src/main/java/rudyAir/model/compte/Reservation.java
package rudyAir.model.compte;

import java.util.Objects;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;
import javax.validation.constraints.DecimalMin;

import com.fasterxml.jackson.annotation.JsonView;

import rudyAir.model.Views;
import rudyAir.model.vol.Vol;

@Entity
@Table(name = "reservation")
@SequenceGenerator(name = "seqReservation", sequenceName = "seq_reservation", initialValue = 100, allocationSize = 1)
public class Reservation {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqReservation")
	@JsonView(Views.Common.class)
	private Long id;
	@JsonView({ Views.Reservation.class, Views.SiegeWithReservationAndAvion.class,
			Views.CompteClientWithReservation.class })
	private boolean statut;
	@DecimalMin("0")
	@JsonView({ Views.Reservation.class, Views.SiegeWithReservationAndAvion.class,
			Views.CompteClientWithReservation.class })
	private Integer animaux;
	@DecimalMin("0")
	@JsonView({ Views.Reservation.class, Views.SiegeWithReservationAndAvion.class,
			Views.CompteClientWithReservation.class })
	private int bagage;
	@JsonView({ Views.Reservation.class, Views.SiegeWithReservationAndAvion.class,
			Views.CompteClientWithReservation.class })
	private String siege;

	@OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
	@JoinColumn(name = "resa_passager_id", foreignKey = @ForeignKey(name = "resa_passager_id_fk"), nullable = false)
	@JsonView({ Views.Reservation.class, Views.SiegeWithReservationAndAvion.class,
			Views.CompteClientWithReservation.class })
	private Passager passager;
	@ManyToOne
	@JoinColumn(name = "resa_vol_id", foreignKey = @ForeignKey(name = "resa_vol_id_fk"), nullable = false)
	@JsonView({ Views.Reservation.class, Views.CompteClientWithReservation.class })
	private Vol vol;

	@ManyToOne
	@JoinColumn(name = "resa_client_id", foreignKey = @ForeignKey(name = "resa_client_id_fk"), nullable = false)
	@JsonView(Views.Reservation.class)
	private Client client;

	@Version
	private int version;

	public Reservation() {
	}

	public Reservation(Long id, Vol vol, Passager passager, Client client, boolean statut,
			@DecimalMin("0") Integer animaux, @DecimalMin("0") int bagage, String siege) {
		super();
		this.id = id;
		this.vol = vol;
		this.passager = passager;
		this.client = client;
		this.statut = statut;
		this.animaux = animaux;
		this.bagage = bagage;
		this.siege = siege;
	}

	public Reservation(Vol vol, Passager passager, Client client, boolean statut, @DecimalMin("0") Integer animaux,
			@DecimalMin("0") int bagage, String siege) {
		super();
		this.vol = vol;
		this.passager = passager;
		this.client = client;
		this.statut = statut;
		this.animaux = animaux;
		this.bagage = bagage;
		this.siege = siege;
	}

	public Reservation(Vol vol, Passager passager, Client client, boolean statut, @DecimalMin("0") Integer animaux,
			@DecimalMin("0") int bagage) {
		super();
		this.vol = vol;
		this.passager = passager;
		this.client = client;
		this.statut = statut;
		this.animaux = animaux;
		this.bagage = bagage;
	}

	public String getSiege() {
		return siege;
	}

	public void setSiege(String siege) {
		this.siege = siege;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Vol getVol() {
		return vol;
	}

	public void setVol(Vol vol) {
		this.vol = vol;
	}

	public Passager getPassager() {
		return passager;
	}

	public void setPassager(Passager passager) {
		this.passager = passager;
	}

	public Integer getAnimaux() {
		return animaux;
	}

	public void setAnimaux(Integer animaux) {
		this.animaux = animaux;
	}

	public int getBagage() {
		return bagage;
	}

	public void setBagage(int bagage) {
		this.bagage = bagage;
	}

	public boolean isStatut() {
		return statut;
	}

	public void setStatut(boolean statut) {
		this.statut = statut;
	}

	public Client getClient() {
		return client;
	}

	public void setClient(Client client) {
		this.client = client;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public String toString() {
		return "Reservation [id=" + id + ", statut=" + statut + ", animaux=" + animaux + ", bagage=" + bagage
				+ ", siege=" + siege + ", passager=" + passager + ", vol=" + vol + ", client=" + client + ", version="
				+ version + "]";
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Reservation other = (Reservation) obj;
		return Objects.equals(id, other.id);
	}

}

File: rudyAirAngular/src/app/model/compte/reservation.ts
import { Vol } from '../vol/vol';
import { Client } from './client';
import { Passager } from './passager';

export class Reservation {
  private _id: number | undefined;
  private _statut: boolean | undefined;
  private _animaux: number | undefined;
  private _bagage: number | undefined;
  private _passager: Passager | undefined;
  private _vol: Vol | undefined;
  private _siege: string | undefined;
  private _client: Client | undefined;

  constructor(
    id?: number,
    statut?: boolean,
    animaux?: number,
    bagage?: number,
    passager?: Passager,
    vol?: Vol,
    siege?: string,
    client?: Client
  ) {
    this._id = id;
    this._statut = statut;
    this._animaux = animaux;
    this._bagage = bagage;
    this._passager = passager;
    this._vol = vol;
    this._siege = siege;
    this._client = client;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter statut
   * @return {boolean }
   */
  public get statut(): boolean | undefined {
    return this._statut;
  }

  /**
   * Getter animaux
   * @return {number }
   */
  public get animaux(): number | undefined {
    return this._animaux;
  }

  /**
   * Getter bagage
   * @return {number }
   */
  public get bagage(): number | undefined {
    return this._bagage;
  }

  /**
   * Getter passager
   * @return {Passager }
   */
  public get passager(): Passager | undefined {
    return this._passager;
  }

  /**
   * Getter vol
   * @return {Vol }
   */
  public get vol(): Vol | undefined {
    return this._vol;
  }

  /**
   * Getter siege
   * @return {string }
   */
  public get siege(): string | undefined {
    return this._siege;
  }

  /**
   * Getter client
   * @return {Client }
   */
  public get client(): Client | undefined {
    return this._client;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter statut
   * @param {boolean } value
   */
  public set statut(value: boolean | undefined) {
    this._statut = value;
  }

  /**
   * Setter animaux
   * @param {number } value
   */
  public set animaux(value: number | undefined) {
    this._animaux = value;
  }

  /**
   * Setter bagage
   * @param {number } value
   */
  public set bagage(value: number | undefined) {
    this._bagage = value;
  }

  /**
   * Setter passager
   * @param {Passager } value
   */
  public set passager(value: Passager | undefined) {
    this._passager = value;
  }

  /**
   * Setter vol
   * @param {Vol } value
   */
  public set vol(value: Vol | undefined) {
    this._vol = value;
  }

  /**
   * Setter siege
   * @param {string } value
   */
  public set siege(value: string | undefined) {
    this._siege = value;
  }

  /**
   * Setter client
   * @param {Client } value
   */
  public set client(value: Client | undefined) {
    this._client = value;
  }
}
